python/ml4ir/base/data/ranklib_to_ml4ir.py

```diff
@@ -0,0 +1,60 @@
+import pandas as pd
+import argparse
+import logging
+
+#Constanst for the column names given to the query id and graded relevance.
+QUERY_ID_NAME = 'qid'
+RELEVANCE_NAME = 'relevance'
+
+
+def process_line(line, keep_additional_info):
+    if keep_additional_info == 1:
+        feature_values = line.replace('#', '').replace(' = ', ':').strip().split()
+    else:
+        feature_values = line.split('#')[0].strip().split()
+    feature_values[0] = RELEVANCE_NAME + ':' + feature_values[0]
+    r = {}
+    for fv in feature_values:
+        feat = fv.split(':')[0].strip()
+        val = fv.split(':')[1].strip()
+        if feat == RELEVANCE_NAME or feat == QUERY_ID_NAME:
+            r[feat] = val
+        else:
+            r['f_' + feat] = val
+    return r
+
+def convert(input_file, output_file, keep_additional_info, gl_2_clicks):
+    f = open(input_file, 'r')
+    rows = []
+
+    for line in f:
+        rows.append(process_line(line, keep_additional_info))
+    f.close()
+
+    df = pd.DataFrame(rows)
+    if gl_2_clicks == 1:
```

Comment:
@mohazahran What is `gl_2_clicks`?

Reply:
Might be useful to add a few comments about what is happening here.

Reply:
it's one of the script's arguments. It's a flag to whether to convert the graded relevance to clicks or not.